src/GraphBLAS-sharp.Backend/Algorithms/Algorithms.fs

```diff
@@ -14,3 +14,8 @@ module Algorithms =
 
     module SSSP =
         let singleSource = SSSP.run
+
+    module PageRank =
+        let run = PageRank.run
+
+        let prepareMatrix = PageRank.prepareMatrix
```

Comment:
I think that `prepareMatrix` should at least have some clarifying comments. Also, to run page rank algorithm user always has to call `prepareMatrix` first. And I don't think he wants to build a lego house every time.

Reply:
Always calling `prepareMatrix` before algorithm isn't necessary. User might already have matrix in correct format, for example in benchmarks. User may also want to obtain the prepared matrix and use it elsewhere.

Reply:
How do we inform the user about this need? What guarantees are there that this condition will be met?

We cannot afford to set such semantic traps at any level of abstraction.

If a precondition arises, it must be expressed in types so that the user is not able to pass an unprepared matrix. 

Otherwise, any mention of functional style should be removed from the title of our work "graphblas in functional style".

Reply:
I believe we do not need to inform user about it because it is not our goal to provide PageRank implementation but to provide enough building blocks to implement it. 
This implementation is nothing more but example and test of our api. 
It is required to know how PageRank works to implement it using our library, so I think we should not care a lot about specific graph algorithms such as BFS or PageRank inside our library

Reply:
@gsvgit What do you think?

Reply:
Ready-to-use garph analysis algorithms should be a part of library. It is the next (possibli highest) leyer of our solution (look at [LaGraph](https://github.com/GraphBLAS/LAGraph)). So, implementations of BFS, PAgeRank and other algorithms should be as clear and documented, as possible. Anyway, even basic implmenetation should be a clear anough to be a starting point for those who want ot develop new algorithms using provieed building blocks.

Regarding types. I agree with @IgorErin: if we can express something using types, we should do it. Ofcourse we should find the best way to do it.